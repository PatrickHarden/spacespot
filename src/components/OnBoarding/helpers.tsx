import {
  OnBoardingState,
  SpaceData,
  SpaceStatus,
  AvailabilityFlexPriceInterval,
} from 'services/onboarding/types'

const date = new Date('2020/01/01')

export const createSpace = (type: string): SpaceData => ({
  id: '0',
  type,
  spaceName: '',
  spaceDescription: '',
  spaceHighlights: '',
  spaceSize: undefined,
  spaceFloor: undefined,
  spaceSizeCommon: undefined,
  spaceMatterPort: '',
  spaceFloored: '',
  use: '',
  spaceRent: undefined,
  spaceServices: undefined,
  spaceServicesNotNegotiable: false,
  availabilityFixed: date,
  months: undefined,
  hotDesks: {
    desks: 0,
    availableFrom: date,
    minLease: 0,
    price: 0,
    currencyCode: 'NOK',
    frequency: AvailabilityFlexPriceInterval.Monthly,
  },
  fixedDesks: {
    desks: 0,
    availableFrom: date,
    minLease: 0,
    price: 0,
    currencyCode: 'NOK',
    frequency: AvailabilityFlexPriceInterval.Monthly,
  },
  servicedOffices: [],
  images: [],
  floorPlan: [],
  fitout: [{ description: 'FitOut1', name: 'FitOut1', amount: 1000 }],
  status: SpaceStatus.ReadyForPublish,
})

export const createOnBoardingState = (): OnBoardingState => ({
  isNew: true,
  buildingId: undefined,
  images: [
    {
      key: 0,
      name: '40VilliersStreet.External_Photo_1_small.png',
      type: 'image/png',
      uri:
        'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP8/5+hHgAHggJ/PchI7wAAAABJRU5ErkJggg==',
    },
  ],
  docs: [],
  name: '',
  description: '',
  address: '',
  city: '',
  postCode: '',
  country: '',
  location: '',
  latLng: undefined,
  district: '',
  subdistrict: {
    id: 0,
    name: '',
  },
  amenities: { hasGym: { checked: true, desc: 'Gym' } },
  spaces: { '0': createSpace('FLEX') },
  spaceCount: 1,
})

export const valInteger = (
  current: number | undefined,
  value: number,
  min: number,
  max?: number,
): number => {
  const val = Number(Number(value).toFixed())
  return val < min || (max && val > max) ? current || 0 : val
}
