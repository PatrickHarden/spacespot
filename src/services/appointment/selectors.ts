import { get } from 'lodash'
import moment from 'moment'
import { SpacespotState } from '../global/types'
import { Appointment } from './types'
import { AppointmentProps } from '@cbreenterprise/spacespot-ui'

import userSelectors from '../user/selectors'

const appointment = (state: SpacespotState) => get(state, 'appointment.data')
const appointmentsUser = (state: SpacespotState) => {
  const appointments = appointment(state)
  const userId = userSelectors.accountIdentifier(state)
  return appointments
    ? appointments.map(
        (appoint: Appointment): AppointmentProps => ({
          id: appoint.id.toString(),
          date: moment(appoint.proposedTime).toDate(),
          description: appoint.description,
          appointmentStatus: appoint.appointmentStatus,
          isOwnLast: userId === appoint.lastProposedBy,
        }),
      )
    : []
}
const error = (state: SpacespotState) => get(state, 'appointment.error')

export default {
  appointment,
  appointmentsUser,
  error,
}
