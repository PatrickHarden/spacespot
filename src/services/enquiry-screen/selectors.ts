import { get } from 'lodash'
import { SpacespotState } from '../global/types'

const getId = (state: SpacespotState) => get(state, 'enquiryscreen.data.id')
const getError = (state: SpacespotState): string =>
  get(state, 'enquiryscreen.error')

export default {
  getId,
  getError,
}
