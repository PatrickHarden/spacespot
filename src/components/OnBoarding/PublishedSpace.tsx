import React, { useState } from 'react'
import { useIntl } from 'react-intl'
import { useDispatch, useSelector } from 'react-redux'
import { get } from 'lodash'
import { FormControlLabel } from '@material-ui/core'
import styled from 'styled-components'
import { TranslatePropSpace } from 'services/space/types'
import {
  formatFloor,
  showAvailability,
  isSpaceFlex,
} from 'services/onboarding/utils'
import {
  getRegionSizeDesc,
  getRegionCurrencyDesc,
} from 'services/global/region'
import SpaceImg from 'assets/img/space.svg'
import history from 'browserhistory'
import { SpaceStatus } from 'services/onboarding/types'
import actions from 'services/onboarding/actions'
import selectors from 'services/dashboard/selectors'
import ConfirmDialog from 'components/common/ConfirmDialog'
import device from 'services/global/device'

const Container = styled.div<{
  error: boolean
  warning: boolean
  clickable: boolean
  deleted: boolean
}>`
  position: relative;
  box-sizing: border-box;
  height: 90px;
  width: 100%;
  border: 1px solid
    ${props =>
      props.error ? '#db4437' : props.warning ? '#EBBF59' : '#dddddd'};
  opacity: 0.98;
  border-radius: 2px;
  background-color: #ffffff;
  display: flex;
  flex-direction: row;
  margin: 10px 0;
  ${props =>
    props.deleted &&
    `
  opacity: 0.2;
  display: none;
  `}
  ${props =>
    props.clickable &&
    `
  :hover {
    box-sizing: border-box;
    border: 1px solid #4fbbd8;
    opacity: 0.98;
    border-radius: 2px;
    box-shadow: 0 0 4px 1px #4fbbd8;
    cursor: pointer;
  }`}
`
const Image = styled.img`
  flex: 0 0 110px;
  width: 110px;
  background-color: #dddddd;
`
const ContainerTitle = styled.div`
  width: calc(100% - 140px);
  padding: 4px 15px;
  @media (min-width: 900px) {
    flex: 0 0 286px;
  }
`
const ContainerInfo = styled.div`
  flex: 0 0 279px;
  margin: 10px 0px;
  padding: 0 10px;
  display: flex;
  justify-content: space-between;
  align-items: flex-start;
  border-left: 1px solid #e7e7e7;
  @media ${device.mobile} {
    display: none;
  }
`

const ContainerControls = styled.div`
  flex: 1 1 270px;
  padding: 4px 15px;
  margin: 10px 0px;
  border-left: 1px solid #e7e7e7;
  display: none;
  @media (min-width: 1000px) {
    display: flex;
    align-items: center;
  }
  label span {
    font-size: 16px;
  }
`

const ControlListed = styled.div`
  flex: 0 0 auto;
`

const ControlEnquiries = styled.div`
  flex: 1 1 auto;
`

const Size = styled.div`
  width: 33.33%;
  padding: 0 5px 0 0;
  :last-child {
    padding: 0;
  }
`
const HotDesks = Size
const FixedDesks = Size
const Rent = Size
const Availability = Size

const ServicedOffices = Availability

const InfoData = styled.div`
  width: 100%;
  color: #000000;
  font-size: 16px;
  font-weight: 400;
`
const SpaceName = styled.div`
  width: 100%;
  color: #000000;
  font-size: 16px;
  font-weight: 500;
  @media ${device.mobile} {
    white-space: nowrap;
    overflow: hidden;
    max-width: 80%;
    text-overflow: ellipsis;
  }
`

const InfoTitle = styled.div`
  width: 100%;
  color: #828286;
  font-size: 14px;
  font-weight: 400;
  word-break: break-all;
`

const InfoDataSub = styled.div`
  margin-top: 4px;
  height: 21px;
  color: #828286;
  font-size: 16px;
  font-weight: 400;
`

const CheckMark = styled((props: { className?: string }) => (
  <span className={props.className}>&#10003;</span>
))`
  color: #4fbbd8;
  font-size: 20px !important;
  padding-right: 4px;
`

const TextWarning = styled.div`
  text-align: right;
  color: #ebbf59;
  font-size: 14px;
  font-weight: 300;
  line-height: 18px;
`
const DeleteButton = styled.button`
  position: absolute;
  top: 4px;
  right: 0px;
  font-size: 10px;
  cursor: pointer;
  border: none;
  z-index: 10;
  background-color: white;
  outline: none;
`

const PublishedSpace = (props: {
  space: TranslatePropSpace
  showControls?: boolean
  clickable?: boolean
}) => {
  const { formatMessage } = useIntl()
  const t = (s: string) => formatMessage({ id: s })
  const dispatch = useDispatch()
  const pendingDelete = useSelector(selectors.pendingDelete)
  const [removePopup, setRemovePopup] = useState(false)
  const { space, showControls } = props
  const isPendingDelete = pendingDelete.includes(space.key)
  const clickable = props.clickable !== false && !isPendingDelete

  const getFirstImage = () => {
    const src = space.photo || get(space, 'images[0].uri') || SpaceImg
    return src
  }
  const status = space.status

  const shorten40 = (name: string) => {
    return name
      ? name.length < 40
        ? name
        : name.substring(0, 36) + '...'
      : name
  }

  const getType = (space: TranslatePropSpace) =>
    isSpaceFlex(space) ? t('ONBOARDING_FLEX') : t('ONBOARDING_FIXED')

  const onClick = () => {
    if (status === SpaceStatus.Published && clickable !== false) {
      history.push(`/onboarding/edit/space/${space.key}`)
    }
  }
  const showError = (
    event: React.MouseEvent<HTMLButtonElement, MouseEvent>,
  ) => {
    event.stopPropagation()
    setRemovePopup(!removePopup)
  }

  const deleteSpace = () => {
    setRemovePopup(false)
    dispatch(actions.deleteSpace(space.key))
  }

  return (
    <>
      <Container
        error={status === SpaceStatus.ErrorPublishing}
        warning={status === SpaceStatus.PublishedWithErrors}
        onClick={onClick}
        clickable={clickable}
        deleted={isPendingDelete}>
        {clickable && (
          <DeleteButton
            data-testid="delete-space"
            title={t('ONBOARDING_REMOVE')}
            onClick={showError}>
            &#9587;
          </DeleteButton>
        )}
        <Image src={getFirstImage()} />
        <ContainerTitle>
          <SpaceName>{shorten40(space.name)}</SpaceName>
          {space.floor !== 0 ? (
            <InfoDataSub>
              {formatFloor(
                space.floor,
                t('ONBOARDING_SPACE_FLOOR'),
                space.country,
              )}
            </InfoDataSub>
          ) : null}
          <InfoDataSub>{getType(space)}</InfoDataSub>
        </ContainerTitle>
        {isSpaceFlex(space) ? (
          <ContainerInfo>
            <HotDesks>
              <InfoData>{space.hotDesks}</InfoData>
              <InfoTitle>{t('ONBOARDING_HOT_DESKS')}</InfoTitle>
            </HotDesks>
            <FixedDesks>
              <InfoData>{space.fixedDesks}</InfoData>
              <InfoTitle>{t('ONBOARDING_FIXED_DESKS')}</InfoTitle>
            </FixedDesks>
            <ServicedOffices>
              <InfoData>{space.sevicedOffices}</InfoData>
              <InfoTitle>{t('ONBOARDING_SERVICED_OFFICES')}</InfoTitle>
            </ServicedOffices>
          </ContainerInfo>
        ) : (
          <ContainerInfo>
            <Size>
              <InfoData>{`${space.size} ${getRegionSizeDesc(
                space.country,
              )}`}</InfoData>
              <InfoTitle>{t('ONBOARDING_TITLE_SIZE')}</InfoTitle>
            </Size>
            <Rent>
              <InfoData>{`${space.rent} ${getRegionCurrencyDesc(
                space.country,
              )}`}</InfoData>
              <InfoTitle>{t('ONBOARDING_MONTHLY_RENT')}</InfoTitle>
            </Rent>
            <Availability>
              <InfoData>
                {showAvailability(space, t('ONBOARDING_AVAILABILITY_NOW'))}
              </InfoData>
              <InfoTitle>{t('ONBOARDING_AVAILABILITY')}</InfoTitle>
            </Availability>
          </ContainerInfo>
        )}
        {showControls && (
          <ContainerControls>
            <ControlEnquiries></ControlEnquiries>
            <ControlListed>
              <FormControlLabel
                control={
                  /*
                <Switch checked={true} color="primary" value="checkedA" />
                */
                  <CheckMark />
                }
                label={t('ONBOARDING_SPACE_LISTED')}
              />
            </ControlListed>
          </ContainerControls>
        )}
      </Container>
      {status === SpaceStatus.PublishedWithErrors ? (
        <TextWarning>
          {t('ONBOARDING_SPACE_PUBLISHED_WITH_WARNING')}
        </TextWarning>
      ) : null}
      {removePopup && (
        <ConfirmDialog
          open={removePopup}
          setOpen={setRemovePopup}
          onClick={deleteSpace}
          title={t('ONBOARDING_REMOVE_SPACE_POPUP_TITLE')}
          cancelText={t('ONBOARDING_REMOVE_SPACE_POPUP_KO')}
          okText={t('ONBOARDING_REMOVE_SPACE_POPUP_OK')}
        />
      )}
    </>
  )
}

export default PublishedSpace
