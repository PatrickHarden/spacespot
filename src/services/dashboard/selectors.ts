import { get, chain } from 'lodash'
import { SpacespotState } from 'services/global/types'

import { Space, TranslatePropSpace } from 'services/space/types'
import selectors from 'services/space/selectors'

import { DashboardState } from './types'

const spacesBuildings = (spaces: Array<Space>): Array<TranslatePropSpace> =>
  spaces
    ? spaces
        .map((space: Space) => selectors.translateBuilding(space, true))
        .sort((spaceA, spaceB) => (spaceA.name > spaceB.name ? 1 : -1))
    : []

const selectDashboard = (state: SpacespotState) => get(state, 'dashboard')

const selectDashboardBuildings = (state: SpacespotState) => {
  const spaces = get(selectDashboard(state), 'buildings', [])
  return spacesBuildings(spaces)
}

const selectDashboardSpaceCurried = (state: SpacespotState) => (id: string) =>
  chain(selectDashboard(state))
    .get('spaces')
    .map((building: Array<Space>) =>
      building.filter((space: Space) => get(space, 'Common.PrimaryKey') === id),
    )
    .filter((building: Array<Space>) => building.length > 0)
    .head()
    .head()
    .value()

const selectDashboardBuildingCurried = (state: SpacespotState) => (
  id: string,
) =>
  chain(selectDashboard(state) as DashboardState)
    .get('buildings')
    .filter((build: Space) => get(build, 'Common.PrimaryKey') === id)
    .head()
    .value()

const selectDashboardSpaces = (state: SpacespotState) => {
  const spaces = get(selectDashboard(state), 'spaces')
  if (!spaces) {
    return {}
  }
  return Object.keys(spaces).reduce(
    (
      accumulator: { [key: string]: Array<TranslatePropSpace> },
      id: string,
    ) => ({
      ...accumulator,
      [id]: spacesBuildings(spaces[id] as Array<Space>),
    }),
    {},
  )
}

const isDashboardLoading = (state: SpacespotState): boolean => {
  return get(state, 'dashboard.isLoading', false)
}

const isDashboardInitOK = (state: SpacespotState): boolean => {
  return get(state, 'dashboard.initOK', false)
}

const pendingDelete = (state: SpacespotState): Array<string> => {
  return get(state, 'dashboard.pendingDelete', [])
}

export default {
  selectDashboardBuildings,
  selectDashboardBuildingCurried,
  selectDashboardSpaceCurried,
  selectDashboardSpaces,
  isDashboardLoading,
  isDashboardInitOK,
  pendingDelete,
}
