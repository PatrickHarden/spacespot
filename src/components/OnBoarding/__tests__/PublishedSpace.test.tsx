import React from 'react'
import { mount } from 'enzyme'
import { mountToJson } from 'enzyme-to-json'
import { createOnBoardingState } from '../helpers'
import selectors from 'services/onboarding/selectors'

import PublishedSpace from '../PublishedSpace'

jest.mock('react-intl', () => ({
  ...jest.requireActual('react-intl'),
  useIntl: () => ({ formatMessage: (s: any) => s.id }),
}))

const mockDate = new Date('2020/01/01')
const mockState = createOnBoardingState()
const mockDispatch = jest.fn()

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: () => mockDispatch,
  useSelector: (selector: (state: any) => any) => selector(mockState),
}))

it('renders without crashing ', async () => {
  const state = { onboarding: mockState }
  const spaces = selectors.onboardingState2Space(state)
  spaces[0].availability = mockDate
  const wrapper1 = mount(
    <PublishedSpace space={spaces[0]} key={spaces[0].key} />,
  )
  expect(mountToJson(wrapper1)).toMatchSnapshot()
})
