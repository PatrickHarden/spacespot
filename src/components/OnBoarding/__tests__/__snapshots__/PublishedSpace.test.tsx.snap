// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing  1`] = `
<PublishedSpace
  space={
    Object {
      "address": "",
      "aspects": null,
      "availability": 2020-01-01T00:00:00.000Z,
      "buildingId": undefined,
      "city": "",
      "country": "NO",
      "description": "",
      "district": "",
      "fixedDesks": 0,
      "floor": 0,
      "hotDesks": 0,
      "images": Array [
        Object {
          "key": 0,
          "name": "40VilliersStreet.External_Photo_1_small.png",
          "type": "image/png",
          "uri": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP8/5+hHgAHggJ/PchI7wAAAABJRU5ErkJggg==",
        },
      ],
      "isNew": false,
      "key": "0",
      "latLng": Object {
        "lat": 0,
        "lng": 0,
      },
      "name": "",
      "postCode": "",
      "rent": "0",
      "sevicedOffices": 0,
      "size": 0,
      "status": 1,
      "subdistrict": Object {
        "id": 0,
        "name": "",
      },
      "type": "FLEX",
    }
  }
>
  <styled.div
    clickable={true}
    deleted={false}
    error={false}
    onClick={[Function]}
    warning={false}
  >
    <StyledComponent
      clickable={true}
      deleted={false}
      error={false}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-gzVnrw",
            "isStatic": false,
            "lastClassName": "gEVIrV",
            "rules": Array [
              "
  position: relative;
  box-sizing: border-box;
  height: 90px;
  width: 100%;
  border: 1px solid
    ",
              [Function],
              ";
  opacity: 0.98;
  border-radius: 2px;
  background-color: #ffffff;
  display: flex;
  flex-direction: row;
  margin: 10px 0;
  ",
              [Function],
              "
  ",
              [Function],
              "
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-gzVnrw",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      onClick={[Function]}
      warning={false}
    >
      <div
        className="sc-gzVnrw gEVIrV"
        onClick={[Function]}
      >
        <styled.button
          data-testid="delete-space"
          onClick={[Function]}
          title="ONBOARDING_REMOVE"
        >
          <StyledComponent
            data-testid="delete-space"
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-kGXeez",
                  "isStatic": false,
                  "lastClassName": "BkEaw",
                  "rules": Array [
                    "
  position: absolute;
  top: 4px;
  right: 0px;
  font-size: 10px;
  cursor: pointer;
  border: none;
  z-index: 10;
  background-color: white;
  outline: none;
",
                  ],
                },
                "displayName": "styled.button",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-kGXeez",
                "target": "button",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            onClick={[Function]}
            title="ONBOARDING_REMOVE"
          >
            <button
              className="sc-kGXeez BkEaw"
              data-testid="delete-space"
              onClick={[Function]}
              title="ONBOARDING_REMOVE"
            >
              â•³
            </button>
          </StyledComponent>
        </styled.button>
        <styled.img
          src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP8/5+hHgAHggJ/PchI7wAAAABJRU5ErkJggg=="
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-htoDjs",
                  "isStatic": false,
                  "lastClassName": "eiQJDY",
                  "rules": Array [
                    "
  flex: 0 0 110px;
  width: 110px;
  background-color: #dddddd;
",
                  ],
                },
                "displayName": "styled.img",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-htoDjs",
                "target": "img",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP8/5+hHgAHggJ/PchI7wAAAABJRU5ErkJggg=="
          >
            <img
              className="sc-htoDjs eiQJDY"
              src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP8/5+hHgAHggJ/PchI7wAAAABJRU5ErkJggg=="
            />
          </StyledComponent>
        </styled.img>
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-dnqmqq",
                  "isStatic": false,
                  "lastClassName": "kYblHQ",
                  "rules": Array [
                    "
  flex: 0 1 286px;
  padding: 4px 15px;
  @media (min-width: 900px) {
    flex: 0 0 286px;
  }
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-dnqmqq",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-dnqmqq kYblHQ"
            >
              <styled.div>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-jzJRlG",
                        "isStatic": false,
                        "lastClassName": "gESKrx",
                        "rules": Array [
                          "
  width: 100%;
  color: #000000;
  font-size: 16px;
  font-weight: 500;
",
                        ],
                      },
                      "displayName": "styled.div",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-jzJRlG",
                      "target": "div",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <div
                    className="sc-jzJRlG gESKrx"
                  />
                </StyledComponent>
              </styled.div>
              <styled.div>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-kAzzGY",
                        "isStatic": false,
                        "lastClassName": "ldvkBz",
                        "rules": Array [
                          "
  margin-top: 4px;
  height: 21px;
  color: #828286;
  font-size: 16px;
  font-weight: 400;
",
                        ],
                      },
                      "displayName": "styled.div",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-kAzzGY",
                      "target": "div",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <div
                    className="sc-kAzzGY ldvkBz"
                  >
                    ONBOARDING_FLEX
                  </div>
                </StyledComponent>
              </styled.div>
            </div>
          </StyledComponent>
        </styled.div>
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-iwsKbI",
                  "isStatic": false,
                  "lastClassName": "bpeSeM",
                  "rules": Array [
                    "
  flex: 0 0 279px;
  margin: 10px 0px;
  padding: 0 10px;
  display: flex;
  justify-content: space-around;
  align-items: center;
  border-left: 1px solid #e7e7e7;
  @media ",
                    "(max-width: 767px)",
                    " {
    display: none;
  }
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-iwsKbI",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-iwsKbI bpeSeM"
            >
              <styled.div>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-jTzLTM",
                        "isStatic": false,
                        "lastClassName": "clQJyL",
                        "rules": Array [
                          "
  flex: 0 0 93px;
  padding: 0 2px;
",
                        ],
                      },
                      "displayName": "styled.div",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-jTzLTM",
                      "target": "div",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <div
                    className="sc-jTzLTM clQJyL"
                  >
                    <styled.div>
                      <StyledComponent
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "sc-fjdhpX",
                              "isStatic": false,
                              "lastClassName": "gQyIjh",
                              "rules": Array [
                                "
  width: 100%;
  color: #000000;
  font-size: 16px;
  font-weight: 400;
",
                              ],
                            },
                            "displayName": "styled.div",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "sc-fjdhpX",
                            "target": "div",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                      >
                        <div
                          className="sc-fjdhpX gQyIjh"
                        >
                          0
                        </div>
                      </StyledComponent>
                    </styled.div>
                    <styled.div>
                      <StyledComponent
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "sc-cSHVUG",
                              "isStatic": false,
                              "lastClassName": "cuFLXO",
                              "rules": Array [
                                "
  height: 21px;
  width: 100%;
  color: #828286;
  font-size: 14px;
  font-weight: 400;
",
                              ],
                            },
                            "displayName": "styled.div",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "sc-cSHVUG",
                            "target": "div",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                      >
                        <div
                          className="sc-cSHVUG cuFLXO"
                        >
                          ONBOARDING_HOT_DESKS
                        </div>
                      </StyledComponent>
                    </styled.div>
                  </div>
                </StyledComponent>
              </styled.div>
              <styled.div>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-jTzLTM",
                        "isStatic": false,
                        "lastClassName": "clQJyL",
                        "rules": Array [
                          "
  flex: 0 0 93px;
  padding: 0 2px;
",
                        ],
                      },
                      "displayName": "styled.div",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-jTzLTM",
                      "target": "div",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <div
                    className="sc-jTzLTM clQJyL"
                  >
                    <styled.div>
                      <StyledComponent
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "sc-fjdhpX",
                              "isStatic": false,
                              "lastClassName": "gQyIjh",
                              "rules": Array [
                                "
  width: 100%;
  color: #000000;
  font-size: 16px;
  font-weight: 400;
",
                              ],
                            },
                            "displayName": "styled.div",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "sc-fjdhpX",
                            "target": "div",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                      >
                        <div
                          className="sc-fjdhpX gQyIjh"
                        >
                          0
                        </div>
                      </StyledComponent>
                    </styled.div>
                    <styled.div>
                      <StyledComponent
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "sc-cSHVUG",
                              "isStatic": false,
                              "lastClassName": "cuFLXO",
                              "rules": Array [
                                "
  height: 21px;
  width: 100%;
  color: #828286;
  font-size: 14px;
  font-weight: 400;
",
                              ],
                            },
                            "displayName": "styled.div",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "sc-cSHVUG",
                            "target": "div",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                      >
                        <div
                          className="sc-cSHVUG cuFLXO"
                        >
                          ONBOARDING_FIXED_DESKS
                        </div>
                      </StyledComponent>
                    </styled.div>
                  </div>
                </StyledComponent>
              </styled.div>
              <styled.div>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-jTzLTM",
                        "isStatic": false,
                        "lastClassName": "clQJyL",
                        "rules": Array [
                          "
  flex: 0 0 93px;
  padding: 0 2px;
",
                        ],
                      },
                      "displayName": "styled.div",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-jTzLTM",
                      "target": "div",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <div
                    className="sc-jTzLTM clQJyL"
                  >
                    <styled.div>
                      <StyledComponent
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "sc-fjdhpX",
                              "isStatic": false,
                              "lastClassName": "gQyIjh",
                              "rules": Array [
                                "
  width: 100%;
  color: #000000;
  font-size: 16px;
  font-weight: 400;
",
                              ],
                            },
                            "displayName": "styled.div",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "sc-fjdhpX",
                            "target": "div",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                      >
                        <div
                          className="sc-fjdhpX gQyIjh"
                        >
                          0
                        </div>
                      </StyledComponent>
                    </styled.div>
                    <styled.div>
                      <StyledComponent
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "sc-cSHVUG",
                              "isStatic": false,
                              "lastClassName": "cuFLXO",
                              "rules": Array [
                                "
  height: 21px;
  width: 100%;
  color: #828286;
  font-size: 14px;
  font-weight: 400;
",
                              ],
                            },
                            "displayName": "styled.div",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "sc-cSHVUG",
                            "target": "div",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                      >
                        <div
                          className="sc-cSHVUG cuFLXO"
                        >
                          ONBOARDING_SERVICED_OFFICES
                        </div>
                      </StyledComponent>
                    </styled.div>
                  </div>
                </StyledComponent>
              </styled.div>
            </div>
          </StyledComponent>
        </styled.div>
      </div>
    </StyledComponent>
  </styled.div>
</PublishedSpace>
`;
